### PR TITLE
Adopt to template stack format

### PR DESCRIPTION
**Pull request by muvaf**
Wordpress stack as template stack.

Note that we lose the `status.endpoint` field on `WordpressInstance` because template stacks do not support backward propagation.

### CODE REVIEW

File: helm-chart/templates/app.yaml
Comment (hasheddan) [inline helm-chart/templates/app.yaml:15]: @muvaf I believe this will need to be a conditional because we don't want to match labels if we are just defaulting to an exiting `KubernetesTarget`

**Review by hasheddan — CHANGES_REQUESTED** [review]

File: helm-chart/templates/app.yaml
Comment (muvaf) [reply helm-chart/templates/app.yaml:15]: Thanks for the heads-up! Just fixed it

**Comment by muvaf** [comment]
I've tested this with `provisionPolicy: ProvisionNewCluster` as it's shown in `example.yaml` and it works with https://github.com/crossplaneio/templating-controller/pull/4 . I'll make this ready for review once that PR is merged.

File: helm-chart/templates/app.yaml
Comment (suskin) [inline helm-chart/templates/app.yaml:5]: Previously, we were able to create multiple wordpresses in the same namespace by using different names in the wordpressinstance CR. Can we still do that if we're using `.Release.Name` for uniqueness?

File: helm-chart/templates/app.yaml
Comment (suskin) [inline helm-chart/templates/app.yaml:7]: Is .Release.Name rendering as expected? With helm2 I was observing that we didn't have access to .Release.Name in the same way as usual.

**Review by suskin — COMMENTED** [review]
We don't necessarily have to do it now, but I want to make a note that we'll probably still want a reference stack which uses controllers, for people who want to use controllers instead of templates. Maybe having the infrastructure stacks will be good enough for that though.

File: helm-chart/templates/app.yaml
Comment (muvaf) [reply helm-chart/templates/app.yaml:5]: Yes, `.Release.Name` is simply the name of the wordpressinstance CR. The only new thing is that the `Namespace` resource in remote cluster is named with `.Release.Name`, too. This is in order to ensure that each WordpressInstance that uses the same cluster deploys a separate namespace so that you can reuse the same cluster with different `WordpressInstance` CRs using `UseExistingTarget`.

File: helm-chart/templates/app.yaml
Comment (muvaf) [reply helm-chart/templates/app.yaml:7]: Yes, it's been working for me.

**Comment by muvaf** [comment]
> We don't necessarily have to do it now, but I want to make a note that we'll probably still want a reference stack which uses controllers, for people who want to use controllers instead of templates. Maybe having the infrastructure stacks will be good enough for that though.

I agree. Maybe a very simple infra stack with only one managed resource and its controller could make sense.

@suskin I've tested this with `crossplane/templating-controller:0.2.0` together with https://github.com/crossplaneio/stack-minimal-gcp/pull/1 . As planned, we'll wait for your testing in dev environment before merging those two adoption PRs.

**Review by hasheddan — APPROVED** [review]
Giving a LGTM based on verifying that when `UseExistingTarget` is provided and
```
targetSelector:
  matchLabels: {}
```
is added that random scheduling to a `KubernetesTarget` in the `Namespace` will occur :+1: 

Ref: https://github.com/crossplaneio/sample-stack-wordpress/pull/31/files#r377842936

**Comment by prasek** [comment]
Will this close out #28?

**Comment by muvaf** [comment]
@prasek looking at the title of https://github.com/crossplaneio/sample-stack-wordpress/issues/28 , yes this PR converts wordpress stack into a template stack. But the description talks about keeping a controller-based wordpress somewhere, which is not something this PR fixes.

**Comment by suskin** [comment]
Similar to what @muvaf said in the [minimal gcp template stack adoption PR](https://github.com/crossplaneio/stack-minimal-gcp/pull/1#issuecomment-586938441), we've verified this in a couple different situations, and template stacks are enabled by default in Crossplane, which means both pre- and post-template stacks from this repo should load properly. I am going to merge this, and if we find any issues during usage or testing in the future, we can work on those as bugs. /cc @muvaf

**Comment by suskin** [comment]
> Will this close out #28?

@prasek What @muvaf said. We'll need to publish a new version and add a new README for it to be closed. Also, it should be fine for us to split the putting up another controller-based wordpress into a separate ticket. We'll still have the older versions of this repository to point to in the meantime.

**Pull request merged**
